### DISCUSSION TITLE
Remove concept of "co-op"/"dual-stage" key modes in mania

### DISCUSSION THREAD

**Discussion by cl8n** [ppy/osu#25936] [Ideas]
I think these just add more complexity to the game for no real benefit.

I recently did a lot of work with the main >10K mania community to help standardize things for RC and so I had to learn a lot about how all of this stuff works. along the way we had many conversations regarding technical problems and poor UX of these key modes, and this is one of the easy problems to isolate and fix imo. I already ran this by them and everyone that responded agreed in some way

what I am suggesting:
- wherever "nK co-op" or "nK + nK" kind of language is used, replace it with "2nK" (e.g. 8K co-op => 16K)
- remove the "dual stage" mod, and replace it with more key mods that extend higher than 10
  - this probably depends on merging key mods together first, or else there are way too many mods lol
- do not use separate keybinds when the stage is split
- leave the choice of splitting the stage entirely up to the user's skin (`SplitStage`). the defaults can match stable so that existing skins don't break (<10K not split, >=10K split)

this downgrades the concept of "dual stage" into purely a skinning one, which is how everyone seems to treat it anyway. it would fix a lot of my initial confusion with these modes, where I didn't understand that my keybinds for the iidx DP style of 16K weren't applying due to my skin using split stages, which is very rare among people who play that regularly. players' skins change in somewhat unexpected ways depending on what keymodes you're playing and whether you are using the co-op ("dual stage" in lazer) mod, which would be fixed here. I also believe the language can be misleading; there's nothing special in how 10K+ keymodes play in osu, it's just more keys.

if this sounds good then I'm also interested in working on it, only after i finish #25796 tho

**Comment by peppy** [comment]
I'd strongly agree with this, from a complexity reduction perspective.

@smoogipoo thoughts?

**Reply by smoogipoo** [reply]
I agree

**Comment by AkiraMorilas** [comment]
I would like to chime in that myself and several other players use split stages exclusively for the feature(?) of having a second set of keybinds. Some even go as far as applying -5 CollumnSpacing on the first collumn after the split to make it look seamless. Attached screenshot is a development version of Kori-'s skin(will link download to the actual skin when I find it)
![screenshot089](https://github.com/ppy/osu/assets/134231050/1f1e0e92-57b1-48e8-8feb-28e3f15c007a)

**Reply by cl8n** [reply]
yeah, as mentioned on discord this is just about the only practical use of it I've seen. i do this too atm for DP vs. EZ2 style in 16K

it seems pretty clear to me that this is taking advantage of a quirk rather than wanting the actual feature, given you are trying to emulate the look of non-split stage with this skin. if my proposed changes are added, I would also advocate for improving 10k+ keybind issues in general (see Konomi's recent suggestions), but I am trying to keep each change set small.

also fwiw the behavior you're showing here does not currently work in lazer anyway.

**Reply by Konomi-io** [reply]
A simple solution to this is to add multiple "keybind sets" per mania keymode that can be easily switched between. It would be something easily ignorable by players who don't need it, but if desired then the player can create more and switch between them like skins.

This is something that can be worried about and tackled in the future if the current idea is implemented though

**Reply by AkiraMorilas** [reply]
Fair point, there's like 4 things I could mention about this screenshot and lazer vs. stable that are out of scope of this idea.
